### DISCUSSION TITLE
Use XMP Crop* fields for the crop/rotation editor

### DISCUSSION THREAD

**Discussion by zdimension** [immich-app/immich#11460] [General]
Really happy that #10989 has been merged! A crop/rotation editor is really useful for an app such as Immich. However I was a bit disappointed to see that it saves to a new image. Of course, editing the original file is not the best approach either since there's value in preserving the original file. 

As such, I was wondering: would it be a good idea to use the Crop{Left, Right, Top, Height, Angle} XMP fields to store that information? They're standard(-ish, they're used by ACR) and they could be stored alongside the original image, in an XMP sidecar. That way we could have some sort of non-destructive editing (possibility of rolling back changes) without having the photo appear twice in the gallery (original + edited).

Such a system (original + sidecar) would mean there'd be two ways of downloading an image, a bit like on Google Photos: download the "visible" image (which would mean, here, to crop and rotate the image server side and return the edited image as a new file) or download the original.

I would be glad to make that into a PR if the idea is deemed good.

**Comment by filski** [comment]
I agree with your suggestion, was the first thing I thought when trying the new crop tool. Especially would be useful as a basic rotate 90/180/270, and XMP works well for other applications like Lightroom etc. 

Could have a similar experience to Lightroom, where if you crop an image with an existing crop, it loads with the existing crop selected.

**Comment by frejaya** [comment]
Agreed! I want to rotate photos to fix how they appear in the timeline, I don't want to create new rotated photos so I have two copies of the same photo with different rotations. I believe sidecar approach accomplishes what I'm looking for. 

The current rotation feature creates an entirely brand new photo with today's date, doesn't keep the original date, so when I'm trying to fix a lot of historical photos it's unfortunately not much help for my timeline. I migrated to Immich before rotation was an option so I've rotated most of my photos outside of Immich before uploading but sometimes one sneaks in that isn't rotated right.

**Comment by zdimension** [comment]
Okay, so after a few tests, it seems like Immich ignores the Orientation field in XMP sidecars. The field value is stored in the `exifInfo` attribute of the asset so changes should be made to the viewers to take that in account when generating the thumbnail and presenting the asset.
